src/datasources/entity-ds-react/queries/queryBuilder.ts

```diff
@@ -1,7 +1,10 @@
 import { TemplateSrv } from '@grafana/runtime'
 import { API } from 'opennms'
+import { EntityQuery, EntityQueryRequest } from './../types'
+import { getAttributeMapping } from './attributeMappings'
 import { EntityTypes } from '../../../constants/constants'
-import { EntityQuery, EntityQueryRequest } from '../types'
+import { getFilterId } from './../../../panels/filter-panel-react/FilterPanelHelper'
```

Comment:
I would recommend to maybe move this functions (FilterEditorData and getFilterId) under a helper class in this directory (datasource) or in a common directory (lib) so they are not depending on a different component.

Reply:
Good recommendation!

I moved data structures mostly into Entity DS, and then moved functions and data structure access into `hooks/useFilterData`.

Now Entity DS has no dependencies on FilterPanel, and FilterPanel can access data and functions from this hook instead of digging into the EntityDS itself.